file from: Work/2018/cart_singleCellInDrop-seq/data_12_10xgenomics/outs/my_output/out201907/ygh_2019_11_20.Rmd

---
title: "R Notebook"
output: html_notebook
---

This is an [R Markdown](http://rmarkdown.rstudio.com) Notebook. When you execute code within the notebook, the results appear beneath the code.

Try executing this chunk by clicking the *Run* button within the chunk or by placing your cursor inside it and pressing *Ctrl+Shift+Enter*.

```{r}
library(Seurat)
library(ggplot2)
library(RColorBrewer)
library(magrittr) # need to run every time you start R and want to use %>%
library(dplyr)
library(plyr)
setwd("g:/work/2018_01/cart_singleCellInDrop-seq/data_12_10xgenomics/outs/my_output/out201907")
cart <- readRDS("1_cart_ygh_intergrated.rds")
```

```{r}
exp <- GetAssayData(object = cart, slot = 'data')
cc.file <- read.table("cell_cycle_genes.txt",head=T)
mygenes <- as.character(unique(cc.file$gene))
mygenes <- intersect(mygenes, rownames(exp))
mydata <- exp[mygenes,]
mycolors = c("#E71918","#357FBC","#45AD4C","#A351A3","#FF7E00","#FFED22","#AD5322","#F99DD1","#fb8072","#59C4A6","#FD8E63","#90A1CF","#9A9A9A","#A6DB4C","#E3C594","#92D3C8","#003c30","#dfc27d","#c51b7d","#053061","#67001f","#E88AC0","#1a1a1a","#17617B")
p <- DimPlot(cart, reduction = "umap", label = TRUE, cols= mycolors,label.size=20, vector.friendly=T,pt.size=0.1)
pbuild <- ggplot2::ggplot_build(p) # Use ggplot_build to deconstruct the ggplot object
pdata <- pbuild$data[[1]] #
metaData <-cart@meta.data
pdata$samples <- metaData$orig.ident
pdata$group <- metaData$seurat_clusters
rownames(pdata) <- rownames(metaData)
head(pdata)
pdata$feature <- colMeans(as.data.frame(mydata))
head(pdata)
pdata2 <- pdata[pdata$samples %in% c("sample1","sample3","sample5"),]
dim(pdata2)     
```

```{r}
ggplot(pdata2, aes(x,y)) +
geom_point(aes(col=feature, size = feature,alpha = feature)) +
     theme_bw() +
     scale_color_gradient(low = "forestgreen", high = "red", trans = "sqrt", limits = c(0,2)) +
     scale_size_continuous(range = c(0.1,0.8)) +
     scale_alpha_continuous(range = c(0.5,1)) +
     guides(size = "none", alpha = "none") +
    theme(axis.line=element_blank(),axis.text.x=element_blank(),
          axis.text.y=element_blank(),axis.ticks=element_blank(),
          axis.title.x=element_blank(),
          axis.title.y=element_blank(),
          legend.position=c(0.1,0.8),
          legend.title = element_blank(),
          legend.text = element_text(size = 20),
          panel.background=element_blank(),
          panel.border=element_blank(),
          panel.grid.major=element_blank(),
          panel.grid.minor=element_blank(),
          plot.background=element_blank())
ggsave("raw_figure/8_intergrated_UMAP_cc_score.png",units="in", dpi=600, device='png',width = 6,height = 6)
```

```{r}
pdata2 <- pdata[pdata$samples %in% c("sample2","sample4","sample6"),]
dim(pdata2)  
ggplot(pdata2, aes(x,y)) +
geom_point(aes(col=feature, size = feature,alpha = feature)) +
     theme_bw() +
     scale_color_gradient(low = "forestgreen", high = "red", trans = "sqrt", limits = c(0,2)) +
     scale_size_continuous(range = c(0.1,0.8)) +
     scale_alpha_continuous(range = c(0.5,1)) +
     guides(size = "none", alpha = "none") +
    theme(axis.line=element_blank(),axis.text.x=element_blank(),
          axis.text.y=element_blank(),axis.ticks=element_blank(),
          axis.title.x=element_blank(),
          axis.title.y=element_blank(),
          legend.position=c(0.1,0.8),
          legend.title = element_blank(),
          legend.text = element_text(size = 20),
          panel.background=element_blank(),
          panel.border=element_blank(),
          panel.grid.major=element_blank(),
          panel.grid.minor=element_blank(),
          plot.background=element_blank(),
          legend.key.size = unit(1, "cm"))
#ggsave("raw_figure/8_intergrated_UMAP_cc_score.png",units="in", dpi=600, device='png',width = 6,height = 6)

```

```{r}
library(ggplot2)
setwd("g:/work/2018_01/cart_singleCellInDrop-seq/data_12_10xgenomics/outs/my_output/out201907")
mydata <- read.table("gx/findmarker/plot_gProfiler_16vs7_negetive.txt",head=T,sep="\t")
mydata$pvalue2 <- -log10(mydata$pvalue)
head(mydata)
mydata$term <- factor(mydata$term, levels  = mydata$term)
pdf("raw_figure/6_enrichment_Reactome_cluster16vs7_negetive.pdf")
p <- ggplot(mydata, aes(x=term,y=pvalue2)) + theme_bw()
p + geom_bar(stat="identity",fill="#E63A31",width=0.8) + coord_flip() +
theme(legend.title = element_blank(),
          panel.background=element_blank(),
          panel.grid.major=element_blank(),
          panel.grid.minor=element_blank(),
          plot.background=element_blank()) +
    labs(y="-log10(P value)", x="")
dev.off()
setwd("g:/work/2018_01/cart_singleCellInDrop-seq/data_12_10xgenomics/outs/my_output/out201907")
mydata2 <- read.table("gx/findmarker/plot_gProfiler_16vs7_positive.txt",head=T,sep="\t")
mydata2$pvalue2 <- -log10(mydata2$pvalue)
head(mydata2)
mydata2$term <- factor(mydata2$term, levels  = mydata2$term)
pdf("raw_figure/6_enrichment_Reactome_cluster16vs7_positive.pdf")
p <- ggplot(mydata2, aes(x=term,y=pvalue2)) + theme_bw()
p + geom_bar(stat="identity",fill="#E63A31",width=0.8) + coord_flip() +
theme(legend.title = element_blank(),
          panel.background=element_blank(),
          panel.grid.major=element_blank(),
          panel.grid.minor=element_blank(),
          plot.background=element_blank()) +
    labs(y="-log10(P value)", x="")
dev.off()

```

```{r}
#### cytokine correlation gene score
library(Seurat)
library(ggplot2)
library(RColorBrewer)
library(magrittr) # need to run every time you start R and want to use %>%
library(dplyr)
library(plyr)
setwd("g:/work/2018_01/cart_singleCellInDrop-seq/data_12_10xgenomics/outs/my_output/out201907")
cart <- readRDS("1_cart_ygh_intergrated.rds")
exp <- GetAssayData(object = cart, slot = 'data')
cor_gzmb <- read.table("3_GZMB_PRF1_correlationGene.txt")
cytotoxic_gene <- na.omit(cor_gzmb[(cor_gzmb$GZMB >= 0.4 | cor_gzmb$PRF1 >= 0.4),])
mydata <- exp[rownames(cytotoxic_gene),]
mycolors = c("#E71918","#357FBC","#45AD4C","#A351A3","#FF7E00","#FFED22","#AD5322","#F99DD1","#fb8072","#59C4A6","#FD8E63","#90A1CF","#9A9A9A","#A6DB4C","#E3C594","#92D3C8","#003c30","#dfc27d","#c51b7d","#053061","#67001f","#E88AC0","#1a1a1a","#17617B")
p <- DimPlot(cart, reduction = "umap", label = TRUE, cols= mycolors,label.size=20, vector.friendly=T,pt.size=0.1)
pbuild <- ggplot2::ggplot_build(p) # Use ggplot_build to deconstruct the ggplot object
pdata <- pbuild$data[[1]] #
metaData <-cart@meta.data
pdata$samples <- metaData$orig.ident
pdata$group <- metaData$seurat_clusters
rownames(pdata) <- rownames(metaData)
head(pdata)
pdata$feature <- colMeans(as.data.frame(mydata))
head(pdata)
pdata2 <- pdata[pdata$samples %in% c("sample1","sample3","sample5"),]
dim(pdata2)   
ggplot(pdata2, aes(x,y)) +
    geom_point(aes(col=feature, size = feature,alpha = feature)) +
     theme_bw() +
     scale_color_gradient(low = "forestgreen", high = "red", trans = "sqrt", limits = c(0,4)) +
     scale_size_continuous(range = c(0.1,0.8)) +
     scale_alpha_continuous(range = c(0.5,1)) +
     guides(size = "none", alpha = "none") +
    theme(axis.line=element_blank(),axis.text.x=element_blank(),
          axis.text.y=element_blank(),axis.ticks=element_blank(),
          axis.title.x=element_blank(),
          axis.title.y=element_blank(),
          legend.position=c(0.1,0.8),
          legend.title = element_blank(),
          legend.text = element_text(size = 20),
          panel.background=element_blank(),
          panel.border=element_blank(),
          panel.grid.major=element_blank(),
          panel.grid.minor=element_blank(),
          plot.background=element_blank(),
          legend.key.size = unit(1, "cm"))
ggsave("raw_figure/8_intergrated_UMAP_cyto_score.png",units="in", dpi=600, device='png',width = 6,height = 6)
```


```{r}
setwd("g:/work/2018_01/cart_singleCellInDrop-seq/data_12_10xgenomics/outs/my_output/out201907")
cytokineTF <- read.csv("gx/dfcytokineTF.csv")
head(cytokineTF)
#    X row.names         p_val  avg_logFC pct.1 pct.2     p_val_adj upordown  from     type
#1   4     HMGB2 5.570961e-266  0.6263074 1.000 0.999 1.030684e-261     down 16vs7       TF
#2 163      ATF3  1.071533e-33 -0.5509074 0.364 0.562  1.982443e-29       up 16vs7       TF
#3 170      KLF2  3.333947e-30  0.4327580 0.590 0.420  6.168136e-26     down 16vs7       TF
#4 181      CCL4  4.385315e-28  0.6373916 0.518 0.340  8.113272e-24     down 16vs7 cytokine
TF <- cytokineTF[cytokineTF$type == "TF",]
TF_gene <- unique(as.character(TF$row.names))
length(TF_gene)
#[1] 34

cart <- readRDS("1_cart_ygh_intergrated.rds")
exp <- GetAssayData(object = cart, slot = 'data')
TF2 <- intersect(TF_gene, as.character(rownames(exp)))
mydata <- as.data.frame(exp[TF2,])
dim(mydata)
#[1]    34 55488
head(mydata[,1:6])
#      sample1_1 sample1_4 sample1_5 sample1_7 sample1_8 sample1_9
#HMGB2  3.748621 2.3478429 1.4449149 4.0162589  1.716714  1.193094
#ATF3   1.796043 1.7459752 0.9634588 0.0000000  0.000000  0.000000
#KLF2   0.000000 0.7806414 0.0000000 0.0000000  0.000000  0.000000
pdata <- cart@meta.data
exp_mean <- aggregate(t(mydata), list(pdata$seurat_clusters), mean)
rownames(exp_mean) <- exp_mean$Group.1
exp_mean <- exp_mean[,-1]
pdata <- exp_mean[rownames(exp_mean) != "17",]
pdata <- pdata[c("4","16","7","6","13","18","12","1","0","20","5","22","2","11","3","9","15","21","14","10","23","8","19"),]

library(pheatmap)
library(viridis)
library(RColorBrewer)
breaksList = seq(-2, 2, by = 0.2)
#breaksList = c(seq(0, 2, by = 0.2),3.5)
pheatmap(t(pdata),
    scale = "row",
    #color = colorRampPalette(rev(brewer.pal(n = 7, name = "RdYlBu")))(length(breaksList)),
    #color = inferno(length(breaksList) - 1),
    #colorRampPalette(c("navy", "white", "firebrick3"))(20),
    colorRampPalette(c("#667FA2", "white", "#D05B5B"))(length(breaksList) - 1),
    breaks = breaksList,
    cluster_cols = F,
    gaps_col = 14)
pdf("raw_figure/7_TF_heatmap.pdf")
```


```{r}
library(Seurat)
library(ggplot2)
library(ggridges)

setwd("g:/work/2018_01/cart_singleCellInDrop-seq/data_12_10xgenomics/outs/my_output/out201907")
cytokine <- read.table(".../../../../../cart_cell_cytokines_add.txt",header = 1, sep = "\t")
head(cytokine)
#  symbol        name     type
#1   GZMB  Granzyme B Effector
#2   IFNG   IFN GAMMA Effector
cart <- readRDS("1_cart_ygh_intergrated.rds")
exp <- GetAssayData(object = cart, slot = 'data')
cytokine2 <- intersect(as.character(cytokine$symbol), as.character(rownames(exp)))
mydata <- as.data.frame(exp[cytokine2,])
dim(mydata)
#[1]    28 55488
head(mydata[,1:6])
#     sample1_1 sample1_4 sample1_5 sample1_7 sample1_8 sample1_9
#GZMB  0.813788  0.000000  1.768531 1.4072032  1.397269         0
#IFNG  0.000000  0.000000  0.000000 0.7071349  0.000000         0
#CCL3  0.000000  0.000000  0.000000 0.0000000  0.000000         0
mycytokine_count <- colSums(!!mydata)
head(mycytokine_count)
#sample1_1 sample1_4 sample1_5 sample1_7 sample1_8 sample1_9
#        3         5         4         2         2         0
pdata <- cart@meta.data
pdata$count_cytokine <- mycytokine_count
pdata2 <- pdata[!pdata$orig.ident %in% c("sample2"),]
table(pdata2$orig.ident)
#sample1 sample3 sample5
#   7204   11848    9464
write.table(pdata2, "5_count_cytokines.txt", sep="\t")
pdata <- read.table("5_count_cytokines.txt")
pdata$orig.ident <- factor(pdata$orig.ident, levels = c("sample6","sample5","sample4","sample3","sample1"))
ggplot(pdata,aes(x=count_cytokine, y=orig.ident, fill=orig.ident)) +
    geom_density_ridges(scale = 2,bandwidth = 0.5,alpha = 0.8) +
    scale_fill_manual(values=c("#f3e96b", "#c2d3da", "#f28a30", "#81a3a7", "#f05837", "#8a8683")) +
    theme_bw() +
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank())
    labs(y="-log10(P value)", x="")

ggplot(pdata,aes(x=count_cytokine, y=orig.ident, fill=orig.ident)) +
    theme_bw() +
    theme(panel.grid.minor=element_blank()) +
    geom_density_ridges(bandwidth = 0.5,alpha = 0.8, stat = "binline", bins = 16, scale = 1.5) +
    scale_fill_manual(values=c("#f3e96b", "#c2d3da", "#f28a30", "#81a3a7", "#f05837", "#8a8683")) +
    scale_x_continuous(breaks=seq(0,15,1))
pdf("raw_figure/8_cytokine_count_density_ridges.pdf")

pdata$orig.ident <- factor(pdata$orig.ident, levels = c("sample1","sample3","sample4","sample5","sample6"))
ggplot(pdata) +
    geom_histogram(aes(x = count_cytokine, y = ..density.., fill = orig.ident), binwidth = 1, color = "grey30") +
    scale_fill_manual(values=c("#f3e96b", "#c2d3da", "#f28a30", "#81a3a7", "#f05837", "#8a8683")) +
    facet_grid(orig.ident ~ .) +
    theme_bw() +
    theme(panel.grid.minor=element_blank())
pdf("raw_figure/8_cytokine_count_density_histogram2.pdf")
```

```{r}
library(Seurat)
library(ggplot2)
library(ggridges)

setwd("g:/work/2018_01/cart_singleCellInDrop-seq/data_12_10xgenomics/outs/my_output/out201907")
cytokine <- read.table(".../../../../../cart_cell_cytokines_add.txt",header = 1, sep = "\t")
head(cytokine)
#  symbol        name     type
#1   GZMB  Granzyme B Effector
#2   IFNG   IFN GAMMA Effector
cart <- readRDS("1_cart_ygh_intergrated.rds")
exp <- GetAssayData(object = cart, slot = 'data')
cytokine2 <- intersect(as.character(cytokine$symbol), as.character(rownames(exp)))
mydata <- as.data.frame(exp[cytokine2,])
dim(mydata)
#[1]    28 55488
head(mydata[,1:6])
#     sample1_1 sample1_4 sample1_5 sample1_7 sample1_8 sample1_9
#GZMB  0.813788  0.000000  1.768531 1.4072032  1.397269         0
#IFNG  0.000000  0.000000  0.000000 0.7071349  0.000000         0
#CCL3  0.000000  0.000000  0.000000 0.0000000  0.000000         0
pdata <- cart@meta.data
#total <- cbind(pdata, t(mydata))
exp_mean <- aggregate(t(mydata), list(pdata$seurat_clusters), mean)
rownames(exp_mean) <- exp_mean$Group.1
exp_mean <- exp_mean[,-1]
cyto <- c("TNFRSF9","TNF","IL22","IL5","CD40LG","IL4","IL13","LTA","CXCL8","TGFB1","GZMB","IFNG","CCL3","PRF1","CCL4","CCL5","IL10")
exp_mean2 <- exp_mean[,colnames(exp_mean) %in% cyto]
pdata <- exp_mean2[rownames(exp_mean2) != "17",]
pdata <- pdata[c("4","16","7","6","13","18","12","1","0","20","5","22","2","11","3","9","15","21","14","10","23","8","19"),]

library(pheatmap)
library(viridis)
library(RColorBrewer)
breaksList = seq(-2, 2, by = 0.2)
annotation_row <- data.frame(row.names=colnames(pdata),sample=c(rep("effector",6),rep("stimulatory",2),rep("chemoattractive",2),rep("regulatory",7)))
anno_colors <- list(sample=c("effector" = "#38AA34", "stimulatory" = "#12497F", "chemoattractive" = "#5C2164", "regulatory" = "#F8BE00"))
pheatmap(t(pdata),
    scale = "row",
    #color = colorRampPalette(rev(brewer.pal(n = 7, name = "RdYlBu")))(length(breaksList)),
    #color = inferno(length(breaksList) - 1),
    #colorRampPalette(c("navy", "white", "firebrick3"))(20),
    colorRampPalette(c("#667FA2", "white", "#D05B5B"))(20),
    annotation_row=annotation_row,
    annotation_colors = anno_colors,
    breaks = breaksList,
    cluster_cols = F,
    gaps_col = 14)
```

```{r}
library(Seurat)
library(ggplot2)
library(ggridges)

setwd("g:/work/2018_01/cart_singleCellInDrop-seq/data_12_10xgenomics/outs/my_output/out201907")
cytokine <- read.table(".../../../../../cart_cell_cytokines_add.txt",header = 1, sep = "\t")
head(cytokine)
#  symbol        name     type
#1   GZMB  Granzyme B Effector
#2   IFNG   IFN GAMMA Effector
cart <- readRDS("1_cart_ygh_intergrated.rds")
exp <- GetAssayData(object = cart, slot = 'data')
cytokine2 <- intersect(as.character(cytokine$symbol), as.character(rownames(exp)))
mydata <- as.data.frame(exp[cytokine2,])
dim(mydata)
#[1]    28 55488
head(mydata[,1:6])
#     sample1_1 sample1_4 sample1_5 sample1_7 sample1_8 sample1_9
#GZMB  0.813788  0.000000  1.768531 1.4072032  1.397269         0
#IFNG  0.000000  0.000000  0.000000 0.7071349  0.000000         0
#CCL3  0.000000  0.000000  0.000000 0.0000000  0.000000         0
pdata <- cart@meta.data
pdata2 <- pdata[pdata$orig.ident %in% c("sample1","sample3","sample5"),]
mydata2 <- mydata[,rownames(pdata2)]
exp_mean <- aggregate(t(mydata2), list(pdata2$seurat_clusters), mean)
rownames(exp_mean) <- exp_mean$Group.1
exp_mean <- exp_mean[,-1]
#cyto <- c("TNFRSF9","TNF","IL22","IL5","CD40LG","IL4","IL13","LTA","CXCL8","TGFB1","GZMB","IFNG","CCL3","PRF1","CCL4","CCL5","IL10")
cyto <- c("TNFRSF9","TNF","IL22","IL5","CD40LG","IL4","IL13","LTA","CXCL8","TGFB1","GZMB","IFNG","CCL3","PRF1","CCL4","CCL5")
exp_mean2 <- exp_mean[,colnames(exp_mean) %in% cyto]
pdata <- exp_mean2[rownames(exp_mean2) != "17",]
#pdata <- pdata[c("4","16","7","6","13","18","12","1","0","20","5","22","2","11","3","9","15","21","14","10","23","8","19"),]
pdata <- pdata[c("4","16","7","13","6","18","12","1","0","20","5","22"),]
library(pheatmap)
library(viridis)
library(RColorBrewer)
breaksList = seq(-2, 2, by = 0.2)
annotation_row <- data.frame(row.names=colnames(pdata),sample=c(rep("effector",6),rep("stimulatory",2),rep("chemoattractive",2),rep("regulatory",6)))
anno_colors <- list(sample=c("effector" = "#38AA34", "stimulatory" = "#12497F", "chemoattractive" = "#5C2164", "regulatory" = "#F8BE00"))
pheatmap(t(pdata),
    scale = "row",
    #color = colorRampPalette(rev(brewer.pal(n = 7, name = "RdYlBu")))(length(breaksList)),
    #color = inferno(length(breaksList) - 1),
    #colorRampPalette(c("navy", "white", "firebrick3"))(20),
    colorRampPalette(c("#667FA2", "white", "#D05B5B"))(20),
    annotation_row=annotation_row,
    annotation_colors = anno_colors,
    breaks = breaksList,
    cluster_cols = F)
pdf("raw_figure/9_cytokine_heatmap_justCART.pdf")

```


```{r}
library(Seurat)
library(ggplot2)
library(RColorBrewer)
library(magrittr) # need to run every time you start R and want to use %>%
library(dplyr)
library(plyr)
setwd("g:/work/2018_01/cart_singleCellInDrop-seq/data_12_10xgenomics/outs/my_output/out201907")
cart <- readRDS("1_cart_ygh_intergrated.rds")
cluster_highcyto.vs.highcc.markers <- FindMarkers(cart,ident.1 = c(6,18,12,1),ident.2 = c(16,7,13), logfc.threshold = 0.25)
cluster_0.vs.highcyto.markers <- FindMarkers(cart,ident.1 = 0,ident.2 = c(6,18,12,1), logfc.threshold = 0.25)
write.table(cluster_highcyto.vs.highcc.markers, file="cluster_highcyto.vs.highcc.markers.txt",quote=F,sep="\t")
write.table(cluster_0.vs.highcyto.markers, file="cluster_0.vs.highcyto.markers.txt",quote=F,sep="\t")
```
```{r}
cd4 <- FeaturePlot(cart, "CD4")
cd8 <- FeaturePlot(cart, "CD8A")
mycolors <- c("#f3e96b", "#c2d3da", "#f28a30", "#81a3a7", "#f05837", "#8a8683")
p <- DimPlot(cart, reduction = "umap", group.by = "orig.ident", cols=mycolors, label.size=20, vector.friendly=T,pt.size=0.1)
pbuild <- ggplot2::ggplot_build(p) # Use ggplot_build to deconstruct the ggplot object
pdata <- pbuild$data[[1]]
metaData <-cart@meta.data
metaData$CD4 <- cd4$data$CD4
metaData$CD8 <- cd8$data$CD8A
head(metaData)
metaData$type = "noneCD4CD8"
metaData[(metaData$CD8 > 0 | metaData$CD4 > 0),11] = "CD4CD8"
metaData[(metaData$CD4 > 0 & metaData$CD8 == 0),11] = "CD4"
metaData[(metaData$CD8 > 0 & metaData$CD4 == 0),11] = "CD8"
table(metaData$orig.ident,metaData$type)
#            CD4 CD4CD8   CD8 noneCD4CD8
#  sample1  2620    129  1381       3074
#  sample2     1      0     3         27
#  sample3   126    149 10414       1159
#  sample4   489    155  4794       2511
#  sample5   217     54  6436       2757
#  sample6   852    201  7276      10663
metaData$cluster_type <- paste(metaData$seurat_clusters, metaData$type, sep = "_")
cart@meta.data <- metaData
cart@meta.data$cluster_sample <- paste(cart@meta.data$seurat_clusters, cart@meta.data$orig.ident, sep = "_")
markers1 <- FindMarkers(cart, ident.1 = c("16_CD4","7_CD4","6_CD4"), ident.2 = c("4_CD4"),group.by = 'type_cluster')
markers2 <- FindMarkers(cart, ident.1 = c("16_sample3","7_sample3","6_sample3"), ident.2 = c("16_sample1","7_sample1","6_sample1"),group.by = 'cluster_sample')
write.table(markers2, "out2020/cluster_highcc_CARTpeak.vs.cluster_highcc_CARTproduct.markers.txt",quote = F, sep = "\t")
```

```{r}

library(Seurat)
library(ggplot2)
library(ggridges)

setwd("g:/work/2018_01/cart_singleCellInDrop-seq/data_12_10xgenomics/outs/my_output/out201907")
cart <- readRDS("1_cart_ygh_intergrated.rds")
exp <- GetAssayData(object = cart, slot = 'data')

ligand_paid <- read.table("gx/ligand_receptor.txt",sep = "\t",header = T)
cyto <- c("TNFRSF9","TNF","IL22","IL5","CD40LG","IL4","IL13","LTA","CXCL8","TGFB1","GZMB","IFNG","CCL3","PRF1","CCL4","CCL5","IL10")
pair_receptor <- ligand_paid[ligand_paid$gene_a %in% cyto,]
receptor <- unique(as.character(pair_receptor$gene_b))
receptor <- receptor[-3]
receptor <- c(receptor, "IL5RA","IL4R","IL13RA1","IL10RA","TNFSF9","IGF2R")

mydata <- as.data.frame(exp[receptor,])
dim(mydata)
#[1]    14 55488
head(mydata[,1:6])
#         sample1_1 sample1_4 sample1_5 sample1_7 sample1_8 sample1_9
#CCR3      0.000000  0.000000 0.0000000         0   0.00000  0.000000
#CCR1      0.000000  0.000000 0.0000000         0   0.00000  0.000000
#TNFRSF1B  0.000000  1.213649 1.4449149         0   0.92508  1.193094
pdata <- cart@meta.data
#total <- cbind(pdata, t(mydata))
exp_mean <- aggregate(t(mydata), list(pdata$seurat_clusters), mean)
rownames(exp_mean) <- exp_mean$Group.1
exp_mean <- exp_mean[,-1]
exp_mean2 <- exp_mean[,colnames(exp_mean) %in% receptor]
pdata <- exp_mean2[rownames(exp_mean2) != "17",]
pdata <- pdata[c("4","16","7","6","13","18","12","1","0","20","5","22","2","11","3","9","15","21","14","10","23","8","19"),]

library(pheatmap)
library(viridis)
library(RColorBrewer)
breaksList = seq(-2, 2, by = 0.2)
#annotation_row <- data.frame(row.names=colnames(pdata),sample=c(rep("effector",6),rep("stimulatory",2),rep("chemoattractive",2),rep("regulatory",7)))
#anno_colors <- list(sample=c("effector" = "#38AA34", "stimulatory" = "#12497F", "chemoattractive" = "#5C2164", "regulatory" = "#F8BE00"))
pheatmap(t(pdata),
    scale = "row",
    #color = colorRampPalette(rev(brewer.pal(n = 7, name = "RdYlBu")))(length(breaksList)),
    #color = inferno(length(breaksList) - 1),
    #colorRampPalette(c("navy", "white", "firebrick3"))(20),
    colorRampPalette(c("#667FA2", "white", "#D05B5B"))(20),
    #annotation_row=annotation_row,
    #annotation_colors = anno_colors,
    breaks = breaksList,
    cluster_cols = F,
    gaps_col = 14)
#pdf("raw_figure/9_cytokine_receptor_heatmap.pdf")

```
```{r}
library(Seurat)
library(ggplot2)
library(ggridges)

setwd("g:/work/2018_01/cart_singleCellInDrop-seq/data_12_10xgenomics/outs/my_output/out201907")
cart <- readRDS("1_cart_ygh_intergrated.rds")
exp <- GetAssayData(object = cart, slot = 'data')
cart_meta <- cart@meta.data$orig.ident %in% c("sample1","sample3","sample5") ## just chose cart cell to calculate correlation
exp_df <- t(as.matrix(exp[,cart_meta]))
head(exp_df[,1:6])
cyto <- c("TNFRSF9","TNF","IL22","IL5","CD40LG","IL4","IL13","LTA","CXCL8","TGFB1","GZMB","IFNG","CCL3","PRF1","CCL4","CCL5","IL10")
exp_cytokine <- exp_df[,cyto]
exp_cytokine <- as.matrix(exp_cytokine)
head(exp_cytokine[,1:8])
#           TNFRSF9      TNF IL22 IL5    CD40LG IL4      IL13      LTA
#sample1_1 1.256436 0.813788    0   0 0.0000000   0 0.0000000 0.000000
#sample1_4 0.000000 0.000000    0   0 0.7806414   0 0.0000000 1.213649
#sample1_5 0.000000 0.000000    0   0 1.7685312   0 0.9634588 0.000000
cor_cytokine <- as.data.frame(t(cor(exp_cytokine, exp_cytokine)))
cor_cytokine[upper.tri(cor_cytokine)] <- NA
melt(cor_cytokine,  na.rm = TRUE)
library(reshape2)
cor_cyto <- melt(as.matrix(cor_cytokine), na.rm = TRUE)
cor_cyto2 <- cor_cyto[abs(cor_cyto$value) >= 0.1 & abs(cor_cyto$value) < 0.9,]
dim(cor_cyto2)
#[1] 38  3
write.table(cor_cyto,file = "out2020/cytokine_cytokine_correlation_network.txt", sep = "\t", quote = F, row.names = F, col.names = T)
```


```{r}
#https://www.cellphonedb.org/explore-sc-rna-seq
# Cellphonedb input data
library(Seurat)
library(ggplot2)
library(ggridges)
library(dplyr)

setwd("g:/work/2018_01/cart_singleCellInDrop-seq/data_12_10xgenomics/outs/my_output/out201907")
cart <- readRDS("1_cart_ygh_intergrated.rds")
exp <- as.data.frame(GetAssayData(object = cart, slot = 'data'))
cart_meta <- cart@meta.data
head(cart_meta)
CAR_meta <- cart_meta[cart_meta$seurat_clusters %in% c(16,7,13,6,18,12,0) & cart$orig.ident %in% c("sample1","sample3","sample5"),]
CAR_meta$sampleID = rownames(CAR_meta)
CAR_meta2 <- as.data.frame(CAR_meta %>% group_by(seurat_clusters) %>% sample_n(500))  ##
head(CAR_meta2)
#  orig.ident nCount_RNA nFeature_RNA condition percent.mt integrated_snn_res.0.5 seurat_clusters integrated_snn_res.1     sampleID
#1    sample5       3432         1187      s5s6   2.797203                      0               0                    0 sample5_2561
#2    sample5       3912         1215      s5s6   2.326176                      0               0                    0 sample5_5301
#3    sample5       3265         1116      s5s6   3.675345                      0               0                    0 sample5_3262
#4    sample5       3498         1156      s5s6   2.773013                      0               0                    0 sample5_1147
dim(CAR_meta)
#[1] 17342     9
dim(CAR_meta2)
#[1] 3500    9
CAR_exp <- exp[,CAR_meta2$sampleID]
dim(CAR_exp)
#[1] 18501  3500
genelist <- read.csv("out2020/cellphonedb_download_gene_input.csv")
head(genelist)
#  gene_name uniprot hgnc_symbol         ensembl
#1      ACE2  Q9BYF1        ACE2 ENSG00000130234
#2     ACKR1  Q16570       ACKR1 ENSG00000213088
mygene <- intersect(genelist$gene_name, rownames(CAR_exp))
length(mygene)
#[1] 624
CAR_exp2 <- CAR_exp[mygene,]
count_file <- data.frame(Gene = rownames(CAR_exp2), CAR_exp2)
write.table(count_file, "out2020/cellphonedb_input_count_file.txt", quote = F, sep = "\t",row.names = F, col.names = T)
dim(CAR_exp2)
#[1]  624 3500
CAR_meta2$type <- "high_cc"
CAR_meta2[CAR_meta2$seurat_clusters %in% c(6,18,12),10] = "high_cyto"
CAR_meta2[CAR_meta2$seurat_clusters == 0,10] = "stable"
meta_file <- CAR_meta2[,9:10]
colnames(meta_file) <- c("Cell","cell_type")
write.table(meta_file, "out2020/cellphonedb_input_meta_file.txt", quote = F, sep = "\t",row.names = F, col.names = T)

### after get cellphonedb PLOT
my
my_mean <- read.table("out2020/cellphonedb/result/means.txt",head = T, sep = "\t")

```


```{r}
## ligand - receptor cellphone output
# Cellphonedb input data
library(ggplot2)
library(ggridges)
library(dplyr)
setwd("g:/work/2018_01/cart_singleCellInDrop-seq/data_12_10xgenomics/outs/my_output/out201907")
mypair <- read.table("gx/cellphoneout/cellphone_ygh_0115/significant_means.txt",header = T, sep = "\t")
mypair_pvalue <- read.table("gx/cellphoneout/cellphone_ygh_0115/pvalues.txt",header = T, sep = "\t")

head(mypair)
colnames(mypair)
mypair[is.na(mypair)] <- 0
#mypair2 <- mypair[,grepl("product_CD4",colnames(mypair))]
mypair2_AB <- cbind(mypair[,c(1:12)], mypair[,c("high_cc.high_cyto","X8.high_cc","X8.high_cyto","X19.high_cc","X19.high_cyto","X11.high_cc","X11.high_cyto")])
mypair2_BA <- cbind(mypair[,c(1:12)], mypair[,c("high_cyto.high_cc","high_cc.8","high_cyto.8","high_cc.19","high_cyto.19","high_cc.11","high_cyto.11")])
mypair3_AB <- mypair2_AB[rowSums(mypair2_AB[,13:19]) > 0,]
mypair3_BA <- mypair2_BA[rowSums(mypair2_BA[,13:19]) > 0,]
rownames(mypair3_AB) <- mypair3_AB$id_cp_interaction
rownames(mypair3_BA) <- mypair3_BA$id_cp_interaction


mypvalue2_AB <- cbind(mypair_pvalue[,c(1:11)], mypair_pvalue[,c("high_cc.high_cyto","X8.high_cc","X8.high_cyto","X19.high_cc","X19.high_cyto","X11.high_cc","X11.high_cyto")])
mypvalue2_BA <- cbind(mypair_pvalue[,c(1:11)], mypair_pvalue[,c("high_cyto.high_cc","high_cc.8","high_cyto.8","high_cc.19","high_cyto.19","high_cc.11","high_cyto.11")])
rownames(mypvalue2_AB) <- mypvalue2_AB$id_cp_interaction
rownames(mypvalue2_BA) <- mypvalue2_BA$id_cp_interaction
mypvalue3_AB <- mypvalue2_AB[rownames(mypair3_AB),]
mypvalue3_BA <- mypvalue2_BA[rownames(mypair3_BA),]
colnames(mypvalue3_BA) <- colnames(mypvalue3_AB)
mypvalue4 <- rbind(mypvalue3_AB,mypvalue3_BA)

head(mypair3_AB)
tmp <- strsplit(as.character(mypair3_BA$interacting_pair),"_")
tmp <- do.call(rbind, tmp)
mypair3_BA$interacting_pair <- paste(tmp[,2],tmp[,1],sep = "_")
colnames(mypair3_BA) <- colnames(mypair3_AB)
mypair4 <- rbind(mypair3_AB,mypair3_BA)

## DOTPLOT mypair4 mypvalue4
library(pheatmap)
library(reshape2)
library(viridis)
library(RColorBrewer)
mypair5 <- melt(mypair4[,c(2,13:19)], id = "interacting_pair")
mypvalue5 <- melt(mypvalue4[,c(2,12:18)], id = "interacting_pair")
pdata <- data.frame(pairgenes = mypair5$interacting_pair, pair_cell = mypair5$variable, pair_means = mypair5$value, pvalue = mypvalue5$value)
head(pdata)
pdata$sig = 5
pdata[pdata$pvalue > 0.05,5] = 0
table(pdata$sig)
p <- ggplot(pdata, aes(x=pair_cell, y=pairgenes, size=sig, color=pair_means)) + geom_point(alpha = 0.8) + theme_classic()
p
p + scale_color_gradient(low = "#FBF138",  high = "#0000FF", space = "Lab", limit = c(0, 2))

## heatmap mypair4
pdata <- mypair4[,c(13,16,17,14,15,18,19)]
rownames(pdata) <- mypair4$interacting_pair
breaksList = c(seq(0, 1.5, by = 0.05),2.3)
#annotation_row <- data.frame(row.names=colnames(pdata),sample=c(rep("effector",6),rep("stimulatory",2),rep("chemoattractive",2),rep("regulatory",7)))
#anno_colors <- list(sample=c("effector" = "#38AA34", "stimulatory" = "#12497F", "chemoattractive" = "#5C2164", "regulatory" = "#F8BE00"))
#the agglomeration method to be used. This should be (an unambiguous abbreviation of) one of "ward.D", "ward.D2", "single", "complete", "average" (= UPGMA), "mcquitty" (= WPGMA), "median" (= WPGMC) or "centroid" (= UPGMC).
pheatmap(pdata,
    #color = colorRampPalette(rev(brewer.pal(n = 7, name = "RdYlBu")))(length(breaksList)),
    #color = inferno(length(breaksList) - 1),
    colorRampPalette(c("white","#9750D5", "#FBF138"))(30),
    #colorRampPalette(c("#667FA2", "white", "#D05B5B"))(30),
    #annotation_row=annotation_row,
    #annotation_colors = anno_colors,
    breaks = breaksList,
    clustering_method = "ward.D",
    cluster_cols = F,
    gaps_col = c(1,3,5))
pdf("raw_figure/10_ligand_receptor_peakCART_timepoint_heatmap.pdf",height = 20)
write.table(pdata,"out2020/ligand_receptor_CARTpeak_timepoint.txt",quote = F, sep = "\t", row.names = T, col.names = T)

pdata_0 <- pdata[pdata$high_cc.high_cyto == 0,]
breaksList = c(seq(0, 1.5, by = 0.05),2.3)
a <- pheatmap(pdata_0,
    colorRampPalette(c("white","#9750D5", "#FBF138"))(30),
    breaks = breaksList,
    clustering_method = "ward.D",
    cluster_cols = F,
    cellheight = 5, cellwidth = 5,
    fontsize = 4,
    gaps_col = c(1,3,5))
pdf("raw_figure/10_ligand_receptor_peakCART_timepoint_heatmap_non_ccTOcyto.pdf",height = 20)

pdata_0 <- pdata[pdata$high_cc.high_cyto > 0,]
breaksList = c(seq(0, 1.5, by = 0.05),2.3)
pheatmap(pdata_0,
    colorRampPalette(c("white","#9750D5", "#FBF138"))(30),
    breaks = breaksList,
    clustering_method = "ward.D",
    cluster_cols = F,
    cellheight = 5, cellwidth = 5,
    fontsize = 4,
    gaps_col = c(1,3,5))
pdf("raw_figure/10_ligand_receptor_peakCART_timepoint_heatmap_ccTOcyto.pdf",height = 20)

### JUST for high_cc state
a$tree_row$order[1:23]
# [1] 30 41 42 43 37 46  6 28 36 29 39  1  2  5 45  4 23 25 32  3 34 21 35
b<- pdata_0[a$tree_row$order[1:23],]
tmp <- strsplit(as.character(rownames(b)),"_")
tmp <- do.call(rbind, tmp)
gene_justTo_highCC <- unique(c(tmp[,1], tmp[,2]))
gene_justTo_highCC <- as.data.frame(c(gene_justTo_highCC, "CD74","APP","TNFRSD14","MIF"))
write.table(gene_justTo_highCC,"out2020/specificGene_FORenrichment/gene_justTo_highCC.txt",quote = F, sep = "\t", row.names = F, col.names = F)

### JUST for cluster19 and 8 state
x <- read.table(file = "clipboard", sep = "_")
b <- unique(c(as.character(x[,1]), as.character(x[,2])))
write.table(b,"out2020/specificGene_FORenrichment/gene_justTo_cluster19cluster8.txt",quote = F, sep = "\t", row.names = F, col.names = F)

### JUST for cluster11 state
x <- read.table(file = "clipboard", sep = "_")
b <- unique(c(as.character(x[,1]), as.character(x[,2])))
write.table(b,"out2020/specificGene_FORenrichment/gene_justTo_cluster11.txt",quote = F, sep = "\t", row.names = F, col.names = F)

### JUST for cc_cyto state
x <- read.table(file = "clipboard", sep = "_")
b <- unique(c(as.character(x[,1]), as.character(x[,2])))
write.table(b,"out2020/specificGene_FORenrichment/gene_justTo_cc_cyto.txt",quote = F, sep = "\t", row.names = F, col.names = F)

### JUST for normalTOcc_cyto state
x <- read.table(file = "clipboard", sep = "_")
b <- unique(c(as.character(x[,1]), as.character(x[,2])))
write.table(b,"out2020/specificGene_FORenrichment/gene_justTo_normalTOcc_cyto.txt",quote = F, sep = "\t", row.names = F, col.names = F)

### JUST for cc-cyto
x <- read.table(file = "clipboard", sep = "_")
b <- unique(c(as.character(x[,1]), as.character(x[,2])))
write.table(b,"out2020/specificGene_FORenrichment/gene_justTo_cc_cyto.txt",quote = F, sep = "\t", row.names = F, col.names = F)

### JUST for NONcc-cyto
x <- read.table(file = "clipboard", sep = "_")
b <- unique(c(as.character(x[,1]), as.character(x[,2])))
write.table(b,"out2020/specificGene_FORenrichment/gene_justTo_NONcc_cyto.txt",quote = F, sep = "\t", row.names = F, col.names = F)

```


```{r}
################ 3_markers.violinPlot.pdf #################
### marker violin Plot
library(Seurat)
library(ggplot2)
library(ggridges)
library(RColorBrewer)
library(RColorBrewer)
library(dplyr)
#mycolors = c("#E71918","#357FBC","#45AD4C","#A351A3","#FF7E00","#FFED22","#AD5322","#F99DD1","#fb8072","#59C4A6","#FD8E63","#90A1CF","#9A9A9A","#A6DB4C","#E3C594","#92D3C8","#003c30","#dfc27d","#c51b7d","#053061","#67001f","#E88AC0","#1a1a1a","#17617B")

setwd("g:/work/2018_01/cart_singleCellInDrop-seq/data_12_10xgenomics/outs/my_output/out201907")
cart <- readRDS("1_cart_ygh_intergrated.rds")
mymarker <- c("CD70","CD27","CD2","CD58","LGALS9","SORL1","CD40","CD44","CD47","HAVCR2","SLC1A5","TNFSF12","TNFRSF25","ICAM1","SPN","ITGAL","CD160","TNFRSF14","IL10","IL10RA","TGFB1","TGFBR3")
mycolor <- c("#66c2a5","#66c2a5","#fc8d62","#fc8d62","#8da0cb","#8da0cb","#8da0cb","#8da0cb","#8da0cb","#8da0cb","#8da0cb","#a6d854","#a6d854","#ffd92f","#ffd92f","#ffd92f","#e5c494","#e5c494","#878787","#878787","#878787","#878787")
for(i in 1:length(mymarker)){
plotdata <- VlnPlot(object = cart, features = mymarker[i])
colnames(plotdata$data)[1] <- "feature"
pdata <- pdata[pdata$feature > 0,]
pdata <- plotdata$data[plotdata$data$ident %in% c(4,16,7,13,6,18,12,1,19,8,11),]
pdata$ident <- factor(pdata$ident, levels = c(4,16,7,13,6,18,12,1,19,8,11))
p <- ggplot(pdata,aes(x=ident, y=feature,fill=mycolor[i])) +
    geom_boxplot(width = 0.4) +
    #scale_fill_manual(values=c("#FF7E00","#003c30","#F99DD1","#A6DB4C","#AD5322","#c51b7d","#9A9A9A","#357FBC","#133059","#EE7B6F","#8C9CC7")) +
    scale_y_continuous(limits = c(0,2)) +
    theme_bw() +
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank())

ggsave(paste(mymarker[i],"pdf", sep = "."), p)
}
```


```{r}
library(Seurat)
library(ggplot2)
library(RColorBrewer)
library(magrittr) # need to run every time you start R and want to use %>%
library(dplyr)
library(plyr)
setwd("g:/work/2018_01/cart_singleCellInDrop-seq/data_12_10xgenomics/outs/my_output/out201907")
cart <- readRDS("1_cart_ygh_intergrated.rds")

cart$newcluster=as.character(cart$integrated_snn_res.1)
cart$newcluster[cart$newcluster%in%c("4")]="cluster4"
cart$newcluster[cart$newcluster%in%c("7","13","16")]="highcc"
cart$newcluster[cart$newcluster%in%c("6","12","18","1")]="highcyto"
cart$cluster4=cart$newcluster
table(cart$newcluster,cart$integrated_snn_res.1)
cart$cluster4[cart$cluster4%in%c("0","20","5")]="stable"
table(cart$cluster4,cart$integrated_snn_res.1)
paradox=subset(cart,cluster4%in%c("cluster4","highcc","highcyto","stable"))
table(paradox$cluster4)
paradox$cluster4 <- factor(paradox$cluster4, levels = c("cluster4","highcc","highcyto","stable"))
Idents(object = paradox) <- paradox$cluster4
library("dplyr")
marker1 <- FindAllMarkers(paradox, assay="RNA",slot="data",only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
top120 <- marker1 %>% group_by(cluster) %>% top_n(n =120, wt = avg_logFC)
write.csv(top120,"out2020/11_top120markerof4cluster.csv")
write.csv(marker1,"out2020/11_Allmarkerof4cluster.csv")

top30 <- marker1 %>% group_by(cluster) %>% top_n(n =30, wt = avg_logFC)

myexp <- GetAssayData(paradox,slot="scale.data",assay="integrated")
genes <- intersect(top30$gene,rownames(myexp))
pdata <- myexp[genes,]



Idents(object = paradox) <- factor(Idents(object = paradox), levels = c("4","high_cc","high_cyto","stable"))
saveRDS(paradox, file = "out2020/12_cart_ygh_intergrated_CART4states.rds")
cart <- readRDS("out2020/12_cart_ygh_intergrated_CART4states.rds")
DoHeatmap(paradox, draw.lines = FALSE, features = intersect(top30$gene,rownames(GetAssayData(paradox,slot="scale.data",assay="integrated"))),assay ="integrated",size=6)+
    theme(axis.text.y= element_text(size=3))
ggsave("11_heatmap.png",units="in", dpi=600, device='png', width = 10,height = 8)


DoHeatmap(paradox, draw.lines = FALSE, features = intersect(top30$gene,rownames(GetAssayData(paradox,slot="scale.data",assay="integrated"))),assay ="integrated",size=6)+
    theme(axis.text.y= element_text(size=3)) +
    scale_fill_gradientn(colors = c("#193166","#f4f4f4", "#9d0623"))
    #scale_fill_gradientn(colors = c("#500015","#f4f4f4", "#1b2c59"))

ggsave("11_heatmap3.png",units="in", dpi=600, device='png')
ggsave("11_heatmap.pdf", device='pdf')
pdf("11_heatmap_top30.pdf")
ggsave("11_heatmap2.png",units="in", dpi=600, device='png')


aa = subset(cart,seurat_clusters == 23)
DoHeatmap(aa, draw.lines = FALSE, features = intersect(top30$gene,rownames(GetAssayData(aa,slot="scale.data",assay="integrated"))),assay ="integrated",size=6)+
    theme(axis.text.y= element_text(size=3))
ggsave("11_heatmap_legend.pdf")


```

```{r}
library(clusterProfiler)
library(org.Hs.eg.db)
library(DOSE)
marker1 <- read.csv("out2020/11_Allmarkerof4cluster.csv",stringsAsFactors = F)
marker <- marker1
id=marker$gene
id1=bitr(unique(id), fromType="SYMBOL", toType="ENTREZID", OrgDb=org.Hs.eg.db, drop = TRUE)
marker1=subset(marker,gene%in%id1$SYMBOL)
marker1$ENTREZID=plyr::mapvalues(marker1$gene,from=id1$SYMBOL,to=id1$ENTREZID)
head(marker1)
#        p_val avg_logFC pct.1 pct.2 p_val_adj  cluster    gene ENTREZID
#TNFRSF4     0  1.455926 0.535 0.066         0 cluster4 TNFRSF4     7293
#ENO1        0  1.366822 0.996 0.860         0 cluster4    ENO1     2023
#PGK1        0  1.350372 0.986 0.662         0 cluster4    PGK1     5230
result1=compareCluster(ENTREZID~cluster, data=marker1, fun='enrichKEGG')
result1 <- as.data.frame(result1)
write.csv(result1,"out2020/13_top120markerof4cluster_enrichment_KEGG2.csv",quote = F)
p1=dotplot(result1,showCategory=5)+theme(axis.text.x = element_text(size = 6,color="black"),axis.text.y = element_text(size = 6,color="black"))


GOresult=compareCluster(ENTREZID~cluster, data=marker1, fun='enrichGO', OrgDb='org.Hs.eg.db',ont="BP")
write.csv(GOresult,"out2020/13_top120markerof4cluster_enrichment_GO.csv")
p2=dotplot(GOresult,showCategory=10,split="cluster")+geom_dotplot(dotsize=0.8)+theme(axis.text.x = element_text(size = 6,color="black"),axis.text.y = element_text(size = 3,color="black"))



###### my ggplot for KEGG ###
library("ggplot2")
setwd("g:/work/2018_01/cart_singleCellInDrop-seq/data_12_10xgenomics/outs/my_output/out201907/")
mydata <- as.data.frame(result1)
mydata <- read.csv("out2020/13_top120markerof4cluster_enrichment_KEGG.csv",row.names = 1,stringsAsFactors = F)
df <- mydata[mydata$Description %in% mydata$Description[c(2:6,20,25:32,34,53,57,58,61,64,66,71,97,101,104,106,107)],]
mixedToFloat <- function(x){
    x <- sub(' ', '+', x, fixed=TRUE)
    return(unlist(lapply(x, function(x) eval(parse(text=x)))))
}
df$GeneRatio_num <- mixedToFloat(as.character(df$GeneRatio))
df$Description <- factor(df$Description, levels=rev(as.character(df$Description)[-c(15:17,19,20,22,23,27:29,32,33)]))
p <- ggplot(df, aes(x=Cluster, y=Description, size=GeneRatio_num, color=p.adjust)) +
    geom_point(alpha = 0.8) +
    theme_bw() +
    theme(panel.grid.major = element_blank(),panel.grid.minor = element_blank()) +
    scale_color_gradient(low = "red2",  high = "mediumblue", space = "Lab", limit = c(2e-42, 0.035)) +
    scale_size(range = c(2, 9))
ggsave("raw_figure/13_enrichment_dotplot2.pdf", p, width = 7, height = 5)

```

```{r}
library(Seurat)
library(ggplot2)
setwd("g:/work/2018_01/cart_singleCellInDrop-seq/data_12_10xgenomics/outs/my_output/out201907")
cart <- readRDS("1_cart_ygh_intergrated.rds")

gene_1_cluster1 <- c("CD3D","CD3G","CD3E","CD4","IL2RA","FOXP3")
gene_2_cluster2 <- c("CD8A","CD8B","GZMB","SELL","CCR7","IL7R")
gene_3_cluster3 <- c("CD69","JUN","FOS","FGFBP2","NKG7","CCL3","CCL5","IFNG")
gene_4_glyco_cellcycle <- c("ENO1","LDHA","TUBB","RRM2","TOP2A","PCNA","MKI67")
gene_5_RP <- c("RPL23","RPL6","RPS27","RPL34","RPL21","RPS25","RPS14","RPL10","RPL3")
gene_6_exhausted <- c("PDCD1","LAG3","TIGIT","HAVCR2")
gene_7_cross1 <- c("LGALS9","CD70","CD27","ICAM1","SPN","ITGAL")
gene_8_cross2 <- c("IL10","IL10RA","TNFSF12","TNFRSF25","CD160","TNFRSF14")

FeaturePlot(cart,features = gene_1_cluster1)
ggsave("raw_figure/featurePlot/ygh_BasicMarkers_gene_1_cluster1.png",units="in", dpi=300, device='png',width = 10,height = 10)

FeaturePlot(cart,features = gene_2_cluster2)
ggsave("raw_figure/featurePlot/ygh_BasicMarkers_gene_2_cluster2.png",units="in", dpi=300, device='png',width = 10,height = 10)

FeaturePlot(cart,features = gene_3_cluster3)
ggsave("raw_figure/featurePlot/ygh_BasicMarkers_gene_3_cluster3.png",units="in", dpi=300, device='png',width = 10,height = 10)

FeaturePlot(cart,features = gene_4_glyco_cellcycle)
ggsave("raw_figure/featurePlot/ygh_BasicMarkers_gene_4_glyco_cellcycle.png",units="in", dpi=300, device='png',width = 10,height = 10)

FeaturePlot(cart,features = gene_5_RP)
ggsave("raw_figure/featurePlot/ygh_BasicMarkers_gene_5_RP.png",units="in", dpi=300, device='png',width = 10,height = 10)

FeaturePlot(cart,features = gene_6_exhausted)
ggsave("raw_figure/featurePlot/ygh_BasicMarkers_gene_6_exhausted.png",units="in", dpi=300, device='png',width = 10,height = 10)

FeaturePlot(cart,features = gene_7_cross1)
ggsave("raw_figure/featurePlot/ygh_BasicMarkers_gene_7_cross1.png",units="in", dpi=300, device='png',width = 10,height = 10)

FeaturePlot(cart,features = gene_8_cross2)
ggsave("raw_figure/featurePlot/ygh_BasicMarkers_gene_8_cross2.png",units="in", dpi=300, device='png',width = 10,height = 10)
```
```{r}
library(Seurat)
library(ggplot2)
setwd("g:/work/2018_01/cart_singleCellInDrop-seq/data_12_10xgenomics/outs/my_output/out201907")
cart <- readRDS("1_cart_ygh_intergrated.rds")

pdata <- FeaturePlot(cart, "TNFRSF4")
colnames(pdata$data)[4] = "feature"
ggplot(pdata$data, aes(UMAP_1,UMAP_2)) +
geom_point(aes(col=feature, size = feature,alpha = feature)) +
     theme_bw() +
     scale_color_gradient(low="grey", high="#910e1a", limits = c(0,4)) +
     scale_size_continuous(range = c(0.001,0.4)) +
     scale_alpha_continuous(range = c(0.005,1)) +
     guides(size = "none", alpha = "none") +
    theme(axis.line=element_blank(),axis.text.x=element_blank(),
          axis.text.y=element_blank(),axis.ticks=element_blank(),
          axis.title.x=element_blank(),
          axis.title.y=element_blank(),
          legend.position=c(0.1,0.8),
          legend.title = element_blank(),
          legend.text = element_text(size = 15),
          panel.background=element_blank(),
          panel.border=element_blank(),
          panel.grid.major=element_blank(),
          panel.grid.minor=element_blank(),
          plot.background=element_blank())
ggsave("raw_figure/1_intergrated_UMAP_marker_TNFRSF4.png",units="in", dpi=600, device='png',width = 6,height = 6)



pdata <- FeaturePlot(cart, "FOS")
colnames(pdata$data)[4] = "feature"
ggplot(pdata$data, aes(UMAP_1,UMAP_2)) +
geom_point(aes(col=feature, size = feature,alpha = feature)) +
     theme_bw() +
     scale_color_gradient(low="grey", high="#910e1a", limits = c(0,5)) +
     scale_size_continuous(range = c(0.001,0.4)) +
     scale_alpha_continuous(range = c(0.005,1)) +
     guides(size = "none", alpha = "none") +
    theme(axis.line=element_blank(),axis.text.x=element_blank(),
          axis.text.y=element_blank(),axis.ticks=element_blank(),
          axis.title.x=element_blank(),
          axis.title.y=element_blank(),
          legend.position=c(0.1,0.8),
          legend.title = element_blank(),
          legend.text = element_text(size = 15),
          panel.background=element_blank(),
          panel.border=element_blank(),
          panel.grid.major=element_blank(),
          panel.grid.minor=element_blank(),
          plot.background=element_blank())
ggsave("raw_figure/1_intergrated_UMAP_marker_FOS.png",units="in", dpi=600, device='png',width = 6,height = 6)

pdata <- FeaturePlot(cart, "RPS27")
colnames(pdata$data)[4] = "feature"
ggplot(pdata$data, aes(UMAP_1,UMAP_2)) +
geom_point(aes(col=feature, size = feature,alpha = feature)) +
     theme_bw() +
     scale_color_gradient(low="grey", high="#910e1a", limits = c(0,6)) +
     scale_size_continuous(range = c(0.001,0.4)) +
     scale_alpha_continuous(range = c(0.005,1)) +
     guides(size = "none", alpha = "none") +
    theme(axis.line=element_blank(),axis.text.x=element_blank(),
          axis.text.y=element_blank(),axis.ticks=element_blank(),
          axis.title.x=element_blank(),
          axis.title.y=element_blank(),
          legend.position=c(0.1,0.8),
          legend.title = element_blank(),
          legend.text = element_text(size = 15),
          panel.background=element_blank(),
          panel.border=element_blank(),
          panel.grid.major=element_blank(),
          panel.grid.minor=element_blank(),
          plot.background=element_blank())
ggsave("raw_figure/1_intergrated_UMAP_marker_RPS27.png",units="in", dpi=600, device='png',width = 6,height = 6)

```

```{r}
library(monocle)
library(Seurat)
library(ggplot2)
setwd("g:/work/2018_01/cart_singleCellInDrop-seq/data_12_10xgenomics/outs/my_output/out201907")
cart <- readRDS("1_cart_ygh_intergrated.rds")
Idents(cart) <- "orig.ident"
#cart <- subset(cart, idents = c("sample1", "sample3","sample5"))
cart <- subset(cart, idents = c("sample3"))

y=cart@meta.data
#y = y[y$orig.ident %in% c("sample1","sample3","sample5"),]
y = y[y$orig.ident %in% c("sample3"),]
expr=GetAssayData(cart,assay="integrated",slot="data")
expr=expr[,rownames(y)]
gene_metadata=data.frame(gene_short_name=rownames(expr))
rownames(gene_metadata)=rownames(expr)
cell_metadata=y
cds <- newCellDataSet(as(expr, "sparseMatrix"),
    phenoData = new("AnnotatedDataFrame", data = cell_metadata),
    featureData = new("AnnotatedDataFrame", data = gene_metadata),
    lowerDetectionLimit = 0.5,
    expressionFamily = negbinomial.size())


var_genes <- cart[["RNA"]]@var.features
ordering_genes <- var_genes

cds <- setOrderingFilter(cds, ordering_genes)
print(dim(exprs(cds)))


seuratX <- cart

#Extract data, phenotype data, and feature data from the SeuratObject
data <- as(as.matrix(seuratX@assays$RNA@data), 'sparseMatrix')

pd <- new('AnnotatedDataFrame', data = seuratX@meta.data)

fData <- data.frame(gene_short_name = row.names(data), row.names = row.names(data))
fd <- new('AnnotatedDataFrame', data = fData)

#Construct monocle cds
HSMM <- newCellDataSet(data,
                              phenoData = pd,
                              featureData = fd,
                              #lowerDetectionLimit = 0.5,
                              expressionFamily = uninormal())# since I have already normalized,

#View data
pData(HSMM)
fData(HSMM)


#Run ordering algorithm
var_genes <- seuratX[["RNA"]]@var.features
ordering_genes <- var_genes

HSMM <- setOrderingFilter(HSMM, ordering_genes)
print(dim(exprs(HSMM)))

## reduce dimension - do not normalize or include pseudo count. Use monocle scaling
HSMM <- reduceDimension(HSMM,norm_method="none",
                        reduction_method="DDRTree",
                        max_components=4,
                        scaling=TRUE,
                        verbose=TRUE,
                        pseudo_expr=0)

# First decide what you want to color your cells by
print(head(pData(HSMM)))
saveRDS(HSMM, file = "1_monocle_cart_sample3.rds")
########
library(monocle)
library(Seurat)
library(ggplot2)
setwd("g:/work/2018_01/cart_singleCellInDrop-seq/data_12_10xgenomics/outs/my_output/out201907")
cart <- readRDS("1_monocle_cart_sample3.rds")
plot_cells(cart, label_groups_by_cluster=FALSE,  color_cells_by = "seurat_clusters")
## order cells change colors and theta to match your plot
options(future.globals.maxSize = 10000 * 1024^2)
memory.limit(size = 10000 * 1024^2)
cart <- orderCells(cart)
plot_cell_trajectory(cart,
                     color_by = "seurat_clusters",
                     theta = -15,
                     show_branch_points = FALSE,
                     show_tree = TRUE,
                     cell_size = 4) + facet_wrap(~seurat_clusters, nrow = 4)
    scale_color_manual(breaks = c("X", "Y", "Z"),
                       values=c("red","green","blue")) +
    theme(legend.position = "right")

# now onwards to pseudotemporal plots
genex <- c("geneX", "geneY")

sig_gene_names <- (genex)
head(sig_gene_names)
pseudotemporalplot <- plot_pseudotime_heatmap(HSMM[sig_gene_names],
                        num_clusters = 9,
                        cores = 4,
                        hmcols = NULL,
                        show_rownames = T)

```

```{r}
## percentage
library(Seurat)
library(ggplot2)
library(plyr)
setwd("g:/work/2018_01/cart_singleCellInDrop-seq/data_12_10xgenomics/outs/my_output/out201907")
cart <- readRDS("1_cart_ygh_intergrated.rds")
mydata <- cart@meta.data
mydata <- mydata[mydata$orig.ident %in% c("sample1","sample3","sample5"),c(1,7)]

pdata <- count(mydata, vars=c("orig.ident","seurat_clusters"))
pdata <- pdata[pdata$orig.ident != "sample2",]
mycolors <- c("#f3e96b", "#f28a30","#f05837")
pdata$seurat_clusters <- factor(pdata$seurat_clusters, levels = c(4,16,7,13,6,18,12,1,0,20,5,22,15,2,11,14,8,19,23,9,3,10,21,17))
ggplot(pdata, aes('', freq, fill = orig.ident)) +
    geom_bar(width = 1, position="fill",stat = "identity", color = "white") +
  facet_grid(". ~ seurat_clusters") +
  coord_polar("y", start = 0)+
  scale_fill_manual(values = mycolors) +
  theme_void()
ggsave("raw_figure/percentage_sample_byCluster.pdf",width = 20,height = 5)

mydata <- cart@meta.data
mydata$type = mydata$seurat_clusters
mydata$type <- as.character(mydata$type)
mydata[mydata$seurat_clusters %in% c(16,7,13),9] = "high_cc"
mydata[mydata$seurat_clusters %in% c(6,18,12,1),9] = "high_cyto"
mydata <- mydata[,c(1,9)]
pdata <- count(mydata, vars=c("orig.ident","type"))
pdata <- pdata[pdata$orig.ident != "sample2",]
mycolors <- c("#f3e96b", "#f28a30", "#81a3a7", "#f05837", "#8a8683")
pdata$type <- factor(pdata$type, levels = c("high_cc",8,4,"high_cc",11,19,2,23,17,3,9,15,20,0,10,22,21,5,14))
ggplot(pdata, aes('', freq, fill = orig.ident)) +
    geom_bar(width = 1, position="fill",stat = "identity", color = "white") +
  facet_grid(". ~ type") +
  coord_polar("y", start = 0)+
  scale_fill_manual(values = mycolors) +
  theme_void()
ggsave("raw_figure/percentage_sample_bycluster2.pdf",width = 20,height = 5)

```{r}
### GEO dataset upload
library(Seurat)
setwd("g:/work/2018_01/cart_singleCellInDrop-seq/data_12_10xgenomics/outs/")
ygh.data.s1 <- Read10X(data.dir="yeguanhuo_S1/filtered_feature_bc_matrix")
ygh.data.s3 <- Read10X(data.dir="yeguanhuo_S3/filtered_feature_bc_matrix")
ygh.data.s4 <- Read10X(data.dir="yeguanhuo_S4/filtered_feature_bc_matrix")
ygh.data.s5 <- Read10X(data.dir="yeguanhuo_S5/filtered_feature_bc_matrix")
ygh.data.s6 <- Read10X(data.dir="yeguanhuo_S6/filtered_feature_bc_matrix")
colnames(ygh.data.s1) <- paste("CART_IP",1:ncol(ygh.data.s1),sep="_")
colnames(ygh.data.s3) <- paste("CART_PP",1:ncol(ygh.data.s3),sep="_")
colnames(ygh.data.s4) <- paste("T_PP",1:ncol(ygh.data.s4),sep="_")
colnames(ygh.data.s5) <- paste("CART_RP",1:ncol(ygh.data.s5),sep="_")
colnames(ygh.data.s6) <- paste("T_RP",1:ncol(ygh.data.s6),sep="_")
ygh.data.s1 <- as.data.frame(ygh.data.s1)
ygh.data.s3 <- as.data.frame(ygh.data.s3)
ygh.data.s4 <- as.data.frame(ygh.data.s4)
ygh.data.s5 <- as.data.frame(ygh.data.s5)
ygh.data.s6 <- as.data.frame(ygh.data.s6)

#setwd("g:/work/2018_01/cart_singleCellInDrop-seq/data_12_10xgenomics/outs/my_output/out201907")
cart <- readRDS("my_output/out201907/1_cart_ygh_intergrated.rds")

ygh.data.s1 <- ygh.data.s1[rownames(cart),]
ygh.data.s3 <- ygh.data.s3[rownames(cart),]
ygh.data.s4 <- ygh.data.s4[rownames(cart),]
ygh.data.s5 <- ygh.data.s5[rownames(cart),]
ygh.data.s6 <- ygh.data.s6[rownames(cart),]

write.csv(ygh.data.s1,"CART_IP.UMI_matrix.csv")
write.csv(ygh.data.s3,"CART_PP.UMI_matrix.csv")
write.csv(ygh.data.s5,"CART_RP.UMI_matrix.csv")
write.csv(ygh.data.s4,"Tcells_PP.UMI_matrix.csv")
write.csv(ygh.data.s6,"Tcells_RP.UMI_matrix.csv")
```
